zebra/zebra_pbr.c

```diff
@@ -542,6 +542,119 @@ void zebra_pbr_del_rule(struct zebra_pbr_rule *rule)
 			   __func__);
 }
 
+void zebra_pbr_process_iptable(struct zebra_dplane_ctx *ctx)
+{
+	int mode, ret;
+	struct zebra_pbr_iptable ipt;
+
+	if (dplane_ctx_get_op(ctx) == DPLANE_OP_IPTABLE_ADD)
+		mode = 1;
+	else
+		mode = 0;
+
+	memset(&ipt, 0, sizeof(ipt));
+	ipt.sock = dplane_ctx_netfilter_get_sock(ctx);
+	ipt.vrf_id = dplane_ctx_get_vrf(ctx);
+	ipt.unique = dplane_ctx_netfilter_get_unique(ctx);
+	ipt.type = dplane_ctx_netfilter_get_type(ctx);
+	ipt.filter_bm = dplane_ctx_netfilter_get_filter_bm(ctx);
+	ipt.fwmark = dplane_ctx_netfilter_get_fwmark(ctx);
+	ipt.action = dplane_ctx_netfilter_get_action(ctx);
+	ipt.pkt_len_min = dplane_ctx_netfilter_get_pktlenmin(ctx);
+	ipt.pkt_len_max = dplane_ctx_netfilter_get_pktlenmax(ctx);
+	ipt.tcp_flags = dplane_ctx_netfilter_get_tcpflags(ctx);
+	ipt.tcp_mask_flags = dplane_ctx_netfilter_get_tcpmaskflags(ctx);
+	ipt.dscp_value = dplane_ctx_netfilter_get_dscpvalue(ctx);
+	ipt.fragment = dplane_ctx_netfilter_get_fragment(ctx);
+	ipt.protocol = dplane_ctx_netfilter_get_protocol(ctx);
+	ipt.flow_label = dplane_ctx_netfilter_get_flowlabel(ctx);
+	ipt.family = dplane_ctx_netfilter_get_family(ctx);
+	ipt.nb_interface = dplane_ctx_netfilter_get_interface_number(ctx);
+	memcpy(&ipt.ipset_name, dplane_ctx_netfilter_get_ipset_name(ctx),
+	       ZEBRA_IPSET_NAME_SIZE);
+
+	if (ipt.nb_interface) {
+		uint32_t i = 0;
+
+		ipt.interface_name_list = list_new();
+		for (i = 0; i < ipt.nb_interface; i++) {
+			listnode_add(
+				ipt.interface_name_list,
+				(void *)dplane_ctx_netfilter_get_interface_val(
+					ctx, i));
+		}
+	}
+	ret = hook_call(zebra_pbr_iptable_update, mode, &ipt);
```

Comment:
I don't see an implementation of this in the code, just these hook calls -  how are these objects being programmed?

Reply:
there was an old pull request proposed that was relying on shell scripts.
obviously, because that was not an ABI API, the pull request was refused, though I was confident.

like what has been done for FPM, I think we have to prepare the zebra_dplane module handle queries that are not purely netlink based. 2 examples:
- this hook script
- as an other subject, zebra dplane should handle "ip forwarding" vty call. today, onlinux we use fopen().. but I think this should be also handled in zebra dplane too.